Clarifications for docs home page (#1575)
tc39/proposal-temporal: docs/README.md

The Temporal API follows a convention of using types whose names start with "Plain" (like `Temporal.PlainDate`, `Temporal.PlainTime`, and `Temporal.PlainDateTime`) for objects which do not have an associated time zone.
Converting between such types and exact time types (`Temporal.Instant` and `Temporal.ZonedDateTime`) can be ambiguous because of time zones and daylight saving time, and the Temporal API lets developers configure how this ambiguity is resolved.

> Temporal documentation also explains core concepts like [exact time, wall-clock time, time zone handling, DST, and ambiguity](./ambiguity.md).

Reviewer: “Temporal documentation also explains” feels slightly confusing to me: Isn’t the current page “Temporal documentation”, too?

Maybe:

```suggestion
Several important concepts are explained elsewhere: [exact time, wall-clock time, time zones, DST, handling ambiguity, and more](./ambiguity.md).
```

* It may additionally make sense to rename `ambiguity.md` to something else (`glossary.md`, `concepts.md`, etc.).
* Alternatively, the section on exact time and wall-clock time could be moved to the main document. Then the file name could stay the same or be changed to `time-zones.md` (etc.).

Author: Suggestion looks good. Applied. I'll leave the URL unchanged because this content is moving to MDN soon anyways, so probably not worth worrying about URLs here. @meyerweb FYI.
